addons/website_event_exhibitor/models/event_event.py

```diff
@@ -25,20 +25,50 @@ def _compute_sponsor_count(self):
         for event in self:
             event.sponsor_count = result.get(event.id, 0)
 
-    @api.depends('event_type_id', 'website_menu', 'exhibitor_menu')
+    @api.depends('event_type_id', 'website_menu')
     def _compute_exhibitor_menu(self):
         for event in self:
             if event.event_type_id and event.event_type_id != event._origin.event_type_id:
                 event.exhibitor_menu = event.event_type_id.exhibitor_menu
-            elif event.website_menu and (event.website_menu != event._origin.website_menu or not event.exhibitor_menu):
+            elif event.website_menu and (event.website_menu != event._origin.website_menu):
                 event.exhibitor_menu = True
             elif not event.website_menu:
                 event.exhibitor_menu = False
 
+    @api.onchange('website_menu')
+    def _on_change_website_menu(self):
+        """use onchange to make sure that website_exhibitor has the correct value
```

Comment:
why not using a compute with store=True ?

Reply:
The previous implementation was using a compute with store=True, the problem with it, is that to propagate the website_menu and event_type changes correctly we need to know which of the two has been changed recently, and when using compute with multiple fields, we cannot tell what field was changed, although we can use _origin for that, but it's not sufficient since _origin cannot be used to determine changes that the user has made when editing the form, for example if the website_menu is unticked then the user ticks it using the form then when he unticks it again the value on origin would be the stored value of the record which is "False" rather then the previous value on the form which is "True", and that is why a single compute method for all fields is not sufficient and why we need several on_change methods in order to be able to handle changes from multiple fields.

Reply:
I still think that onchanges are still needed, because as I said `_origin` does not take into account changes that occur when the user is editing the form, to better understand this, here is a video https://www.awesomescreenshot.com/video/3410453?key=9d457eb47a1fc5da37b4cbad525b2705, of what happens with the proposed solution which only relies on oncompute, the video illustrates well the problem with `_origin`, notice how:

1) When we have "my template" as the event template and then change it to "conferences" the `website_menu` and `website_exhibitor_menu` boxes get unticked and then when we change the event template and go back to "my template" the checkboxes are still unticked evnethough in "my template" they are ticked, this is because `event.event_type_id != event._origin.event_type_id` returns `False` as `event._origin.event_type_id `has the old value of the event_type (the one before the user clicked on edit) which is equals to "my template"

2) When we have "conferences" set as the event_template, then we change to "my template", and then untick `website_menu` and then tick it again, you can notice that `website_exhibitor_menu` is still ticked (it should not be), this is because this time when you untick website_menu `event.event_type_id != event._origin.event_type_id` returns `True`.

> Regarding the event_type onchange, that is indeed more tricky. But I think we can modify the _compute_website_menu instead and take the assumption that no matter the event_type you choose (a new one, come back to the same than before or even re selecting the same that was already set), we take always the website_menu value from the event_type, and too bad if you changed the website menus config on the event before changing the event_type. Event_type values always wins, unless you change the config on the event AFTER.

Let's see what @LNAodoo has to say about this, if this gets approved I think that we should also do the same for the other fields like website_track_menu, website_exhibitor_menu, this would greatly reduce code complexity and in this case we won't need onchanges, only an oncompute will suffice.

> open a event with 'Website Submenu' to False and that has a template that has no website_menu activated (like Exhibition template), activate 'showcase Exhibitor', than activate 'Website Submenu'. You lose the 'Showcase Exhibitor' you just activated.

That is correct, if your proposition is rejected, I think that the best way to solve this is to hide (like we just did in the event_template form) the website_exhibitor_menu field or set its value to false and make it readonly, when website_menu is deactivated, as it does not make much sense to allow the user to activate this field when website_menu is unticked, do you agree with this?

Reply:
> That is correct, if your proposition is rejected, I think that the best way to solve this is to hide (like we just did in the event_template form) the website_exhibitor_menu field or set its value to false and make it readonly, when website_menu is deactivated, as it does not make much sense to allow the user to activate this field when website_menu is unticked, do you agree with this?

@nounoubensebia : I agree, But let's see what @LNAodoo thinks about that :)

For the rest, ok for me